docker.nix

```diff
@@ -0,0 +1,99 @@
+{ dockerTools, backend, frontend, runCommand, writeText, writeScriptBin, bash, cacert, gettext, iana_etc, caddy }:
+let
+  inherit (dockerTools)
+    buildImage
+    buildLayeredImage
+    pullImage
+    shadowSetup;
+
+  inherit (builtins) toJSON;
+
+  httpPort = 80;
+  httpsPort = 443;
+
+  # Some system files necessary for nginx to function normally
+  shadow-files = runCommand "shadow-files" {} ''
+    mkdir -p $out/etc
+    echo "nobody:x:65534:65534:nobody:/:/sbin/nologin" > $out/etc/passwd
+    echo "nobody:!x:::::::" > $out/etc/shadow
+    echo "nobody:x:65534:" > $out/etc/group
+    echo "nobody:x::" > $out/etc/gshadow
+  '';
+
+  backend-config-template = writeText "backend-config-env.yaml.template" (toJSON {
+    api.listen-addr = "*:$API_PORT";
+    api.serve-docs = "$SERVE_DOCS";
+  });
+
+  backend-entry-point = writeScriptBin "entrypoint.sh" ''
+    #!/bin/bash
+    set -euxo pipefail
+
+    /bin/envsubst '$API_PORT $SERVE_DOCS' \
+      < ${backend-config-template} >| /config.yaml
+
+    exec /bin/edna-server -c -c /config.yaml "$@"
+  '';
+
+  caddy-config = writeText "Caddyfile" ''
+    {$DNS_DOMAIN} {
+      gzip
+      root ${frontend}
+
+      proxy /api http://{$API_HOST}:{$API_PORT} {
+        transparent
+      }
+
+      rewrite /static/ {
+        r (.*)
+        to /{1}
+      }
+
+      rewrite {
+        if {path} not_starts_with /api/
+        if {path} not_starts_with /static/
+        to /
+      }
+    }
+  '';
+in
+{
+  backend-image = buildLayeredImage {
+    name = "ghcr.io/serokell/edna-backend";
```

Comment:
Is it the place where the image will be uploaded?

Reply:
Well, yes. But also, no. The image name must match the registry URI where it will be available. This name does not necessarily imply that it is being pushed there now. It's just a name.

Reply:
I don't think this name _must_ match the registry URI.